pkg/filehandler/filefactory.go

```diff
@@ -1,14 +1,27 @@
 package filehandler
 
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/leonidasdeim/goconfig/internal/files"
+)
+
 type FileType string
 
-type fileIO interface {
+type FileIO interface {
 	Write(data any, file string) error
 	Read(data any, file string) error
 	GetExtension() string
 }
 
-func fileIOFactory(t FileType) fileIO {
+func FileIOFactory(o *Optional) FileIO {
```

Comment:
This is actually method and not struct so it is better to give name for it that describes action (verb), like:
```
BuildFileIO
```